<a href="https://colab.research.google.com/github/Iamthulasiraman/Electric_vehicle_Segmentation/blob/main/Ev_EDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
import plotly.express as px


# ***Let analyse the Top EV Based on their features***

### importing ev_cars dataset

In [ ]:
df1 = pd.read_csv('/content/EV_Cars_India_2023_cleaned.csv')
df1

Car_name transmission  Car_price(rs)  Battery_Capacity(KwH)  \
0             MG Comet EV    Automatic         798000                  17.30   
1           Tata Tiago EV    Automatic        1199000                  19.20   
2           Tata Tigor EV    Automatic        1375000                  26.00   
3                MG ZS EV    Automatic        2740000                  50.30   
4              BYD Atto 3    Automatic        3449000                  60.48   
5   Hyundai Kona Electric    Automatic        2403000                  39.20   
6      Mahindra XUV400 EV    Automatic        1898999                  34.50   
7       Tata Nexon EV Max    Automatic        1954000                  40.50   
8       Mercedes-Benz EQS    Automatic       24000000                  17.30   
9          Audi e-tron GT    Automatic       17000000                  19.20   
10                 BMW i4    Automatic        7750000                  26.00   
11                 BYD E6    Automatic        2915000                  50.30   
12         Porsche Taycan    Automatic       23000000                  60.48   
13      Audi RS e-tron GT    Automatic       19000000                  39.20   
14      Mahindra E Verito    Automatic         946000                  34.50   
15        Strom Motors R3    Automatic         450000                  40.50   
16    Tata Nexon EV Prime    Automatic        1719000                  71.70   
17        Hyundai IONIQ 5    Automatic        4495000                  29.20   
18                Kia EV6    Automatic        6595000                  77.40   
19                 BMW iX    Automatic       12000000                  76.60   
20                 BMW i7    Automatic       19000000                 101.70   
21         Mini Cooper SE    Automatic        5250000                  32.60   
22            Audi e-tron    Automatic       12000000                  77.40   
23            Citroen eC3    Automatic        1276000                  29.20   
24    Volvo XC40 Recharge    Automatic        5690000                  78.00   
25          Jaguar I-Pace    Automatic       12000000                  32.60   
26           Pravaig DEFY    Automatic        3950000                 107.80   
27              PMV EaS E    Automatic       47900000                  93.40   
28      Mercedes-Benz EQB    Automatic        7450000                  66.50   
29      Mercedes-Benz EQC    Automatic        9950000                  83.90   

    Drive_range(KMs)  Power(Bhp)  Charge_time(Hours)  
0              230.0       41.42                7.00  
1              250.0       60.34                1.00  
2              315.0       73.75                7.50  
3              461.0      174.33                9.00  
4              521.0      201.15               10.00  
5              452.0      134.10                6.16  
6              375.0      148.00                8.00  
7              453.0      141.04               15.00  
8              230.0       41.42                7.00  
9              250.0       60.34                1.00  
10             315.0       73.75                7.50  
11             461.0      174.33                9.00  
12             521.0      201.15               10.00  
13             452.0      134.10                6.16  
14             375.0       95.00                8.00  
15             453.0      141.04               15.00  
16             415.0       93.87                1.50  
17             320.0       56.22               10.50  
18             708.0      225.86                1.00  
19             425.0      321.84                7.25  
20             625.0      536.40                8.00  
21             270.0      181.03                2.50  
22             708.0      225.86                1.00  
23             425.0      321.84                7.25  
24             625.0      536.40                8.00  
25             270.0      181.03                2.50  
26             857.0      516.29                8.0

In [ ]:
prices= []
for i in df1.sort_values(by = 'Car_price(rs)', ascending = True).values:
  prices.append(i[0])

### Top 10 Costliest Electric vehicle

In [ ]:
price = []
for i in df1.sort_values(by = 'Car_price(rs)', ascending = False).values:
  price.append([i[0],i[2]])

price = price[:10]

In [ ]:
a = pd.DataFrame(price,columns= ['car_name','price'])

In [ ]:
px.bar(a,x='car_name',y = 'price',color = 'car_name',
       title = 'top 10 costliest Ev Cars')

### Top 10 Electric vehicle with maximum Range

In [ ]:

range = []
for i in df1.sort_values(by = 'Drive_range(KMs)', ascending = False).values:
  range.append(i[0])

In [ ]:
b = df1.sort_values(by = 'Drive_range(KMs)', ascending = False)
px.pie(b[:10],names = 'Car_name', values = 'Drive_range(KMs)', title = 'top 10 Cars with maximum Range')

### Top 10 Electric vehicle with less Charging Time

In [ ]:

charge = []
for i in df1.sort_values(by = 'Charge_time(Hours)', ascending = True).values:
  charge.append(i[0])

In [ ]:
a = df1.sort_values(by = 'Charge_time(Hours)', ascending = True)
px.bar(a[:10],x = 'Car_name', y = 'Charge_time(Hours)',color ='Charge_time(Hours)',
       labels ={'x ': 'Car Name', 'y' : 'Charging Time'},
       title = 'top 10 cars taking less time for Charging')

### Top 10 Electric vehicle with maximum Battery capacity



In [ ]:
battery = []
for i in df1.sort_values(by = 'Battery_Capacity(KwH)', ascending = True).values:
  battery.append(i[0])

In [ ]:
a = df1.sort_values(by = 'Battery_Capacity(KwH)', ascending = False)
px.bar(a[:10],x = 'Car_name', y = 'Battery_Capacity(KwH)',color ='Car_name',
       title = 'top 10 cars with maximum Battery Capacity')

### Top 5 Best cars based on their feature and their price

In [ ]:
best_cars = []

for car in prices:
    if car in range and car in battery and car in charge:
        best_cars.append(car)

best_cars[:5]

['Strom Motors R3',
 'MG Comet EV',
 'Mahindra E Verito',
 'Tata Tiago EV',
 'Citroen eC3']

In [ ]:
df1[df1['Car_name'] == 'Strom Motors R3']

Car_name transmission  Car_price(rs)  Battery_Capacity(KwH)  \
15  Strom Motors R3    Automatic         450000                   40.5   

    Drive_range(KMs)  Power(Bhp)  Charge_time(Hours)  
15             453.0      141.04                15.0

In [ ]:
df1[df1['Car_name'] ==  'MG Comet EV']

Car_name transmission  Car_price(rs)  Battery_Capacity(KwH)  \
0  MG Comet EV    Automatic         798000                   17.3   

   Drive_range(KMs)  Power(Bhp)  Charge_time(Hours)  
0             230.0       41.42                 7.0

In [ ]:
df1[df1['Car_name'] ==  'Mahindra E Verito']

Car_name transmission  Car_price(rs)  Battery_Capacity(KwH)  \
14  Mahindra E Verito    Automatic         946000                   34.5   

    Drive_range(KMs)  Power(Bhp)  Charge_time(Hours)  
14             375.0        95.0                 8.0

In [ ]:
df1[df1['Car_name'] ==  'Tata Tiago EV']

Car_name transmission  Car_price(rs)  Battery_Capacity(KwH)  \
1  Tata Tiago EV    Automatic        1199000                   19.2   

   Drive_range(KMs)  Power(Bhp)  Charge_time(Hours)  
1             250.0       60.34                 1.0

In [ ]:
df1[df1['Car_name'] ==  'Citroen eC3']


Car_name transmission  Car_price(rs)  Battery_Capacity(KwH)  \
23  Citroen eC3    Automatic        1276000                   29.2   

    Drive_range(KMs)  Power(Bhp)  Charge_time(Hours)  
23             425.0      321.84                7.25

# **CONCLUSION :**
* **PMV EaSE** is the costliest Electric vehicle
* **Pravaig DEFY** is the ev with  maximum range
* **Audi e-trone GT** is the ev which takes less time for charging
* **Pravaig DEFY** is the ev with maximum Battery Capacity

## Top 5 Ev based on their features and with Low Price


*  Strom Motors R3
*  MG Comet EV
*  Mahindra E Verito
*  Tata Tiago EV
*  Citroen eC3





# ***Lets analyse the availability of charging stations in india***

### importing the charging station Dataset

In [ ]:
df2 = pd.read_csv('/content/RS_Session_257_AU_2368_B_3.csv')
df2

Sl. no.         State/UT         City  \
0         1           Delhi        Delhi   
1         2     Maharashtra  Navi Mumbai   
2         3     Maharashtra       Nagpur   
3         4      Tamil Nadu      Chennai   
4         5          Kerala     Thrissur   
5         6          Kerala    Ernakulum   
6         7          Kerala       Kannur   
7         8         Gujarat    Ahmedabad   
8         9       Karnataka    Bangalore   
9        10  Madhya Pradesh       Indore   
10       11       Rajasthan       Jaipur   
11    Total           Total        Total   

    Operational Charging stations Under FAME-II  
0                                            15  
1                                             1  
2                                             7  
3                                             8  
4                                             8  
5                                             6  
6                                             2  
7                                             2  
8                                             1  
9                                             2  
10                                            1  
11                                           53

### Number of charging station in each city

In [ ]:
df2 = df2.drop(labels = 11, axis = 0)
df2 = df2.sort_values(by = 'Operational Charging stations Under FAME-II', ascending = False)

In [ ]:
px.bar(df2,x='City',y='Operational Charging stations Under FAME-II', color = 'City',
       title = 'Number of charging station in each city :')

### Number of charging station in each state:

In [ ]:
px.bar(df2,x='State/UT',y='Operational Charging stations Under FAME-II', color = 'State/UT',
       title = 'Number of charging station in each state: ')

# **CONCLUSION :**
## City wise comparison :

*   Capital city  Delhi has more number of charging stations
*   Then Chennai and thrissur has the secand more number of charging stations
* nagpur has the third more number of charging stations

## State wise comparison:


*   kerala has more number of charging stations
*   Delhi has secand more number of charging staions

## Overall :
South Indian states like kerala and Tamilnadu have more number of changing stations compared to north india






# ***Lets analyse the sales of petrol and diesel cars***

### importing the cars dataset

In [ ]:
df3 = pd.read_csv('/content/car_data.csv')
df3

Car_Name  Year  Selling_Price  Present_Price  Kms_Driven Fuel_Type  \
0       ritz  2014           3.35           5.59       27000    Petrol   
1        sx4  2013           4.75           9.54       43000    Diesel   
2       ciaz  2017           7.25           9.85        6900    Petrol   
3    wagon r  2011           2.85           4.15        5200    Petrol   
4      swift  2014           4.60           6.87       42450    Diesel   
..       ...   ...            ...            ...         ...       ...   
296     city  2016           9.50          11.60       33988    Diesel   
297     brio  2015           4.00           5.90       60000    Petrol   
298     city  2009           3.35          11.00       87934    Petrol   
299     city  2017          11.50          12.50        9000    Diesel   
300     brio  2016           5.30           5.90        5464    Petrol   

    Seller_Type Transmission  Owner  
0        Dealer       Manual      0  
1        Dealer       Manual      0  
2        Dealer       Manual      0  
3        Dealer       Manual      0  
4        Dealer       Manual      0  
..          ...          ...    ...  
296      Dealer       Manual      0  
297      Dealer       Manual      0  
298      Dealer       Manual      0  
299      Dealer       Manual      0  
300      Dealer       Manual      0  

[301 rows x 9 columns]

In [ ]:
df3['Fuel_Type'].unique()

array(['Petrol', 'Diesel', 'CNG'], dtype=object)

### Sales of car based on type of fuel

In [ ]:
px.bar(df3.groupby('Fuel_Type')['Fuel_Type'].count(),y= 'Fuel_Type' ,color = 'Fuel_Type',
       title = 'sales of car based on fuel type')

In [ ]:
df3[df3['Fuel_Type'] == 'Petrol']['Present_Price'].max()

23.73

In [ ]:
df3[df3['Fuel_Type'] == 'Diesel']['Present_Price'].max()

92.6

### Price of car based on type of fuel

In [ ]:
px.bar(df3,y = list(df3['Selling_Price']),x= list(df3['Fuel_Type']),color = 'Selling_Price',
       labels = {'x':'type of fuel','y':'price of vehicle'}, title = 'Price of car based on type of fuel')

### Plotting the Transmission type:

In [ ]:
px.pie(df3,names = 'Transmission', title = 'type of transmission')

# **CONCLUSION :**


*   People perfers **Petrol cars** than other cars.
*  **Diesel car** is costly than Petrol car.
*   People prefers **Automatic Transmission cars** than Manual Cars
* Least number of people likes to buy CNG cars


# ***Lets analyse the count of ev's in each State***

In [ ]:
df4 = pd.read_csv('/content/geoMap.csv')
df4

Region  no_of_EV
0                    Tamil Nadu     100.0
1                     Karnataka      97.0
2                       Haryana      88.0
3                           Goa      85.0
4                   Maharashtra      85.0
5                    Puducherry      84.0
6                     Telangana      81.0
7                        Kerala      81.0
8                       Gujarat      74.0
9        Dadra and Nagar Haveli      71.0
10               Andhra Pradesh      67.0
11                       Sikkim      67.0
12                       Odisha      65.0
13                 Chhattisgarh      64.0
14                  Uttarakhand      64.0
15  Andaman and Nicobar Islands      62.0
16                    Jharkhand      61.0
17                      Tripura      55.0
18                        Delhi      48.0
19             Himachal Pradesh      47.0
20                    Meghalaya      45.0
21                    Rajasthan      43.0
22                Uttar Pradesh      43.0
23            Arunachal Pradesh      42.0
24                     Nagaland      40.0
25               Madhya Pradesh      40.0
26                       Punjab      38.0
27              Jammu & Kashmir      38.0
28                  West Bengal      34.0
29                        Assam      33.0
30                      Mizoram      33.0
31                      Manipur      29.0
32                        Bihar      24.0
33                   Chandigarh      23.0
34                Daman and Diu       NaN
35                  Lakshadweep       NaN

In [ ]:
px.bar(df4, x= 'Region', y = 'no_of_EV', color = 'Region', title = "Number of ev's in each state")

# **CONCLUSION :**


*   Tamil nadu has more number of Electric vehicles in india
*   Karnataka has secand more number of Electric vehicles in india
*   Lakshadweep and Daman and Diu has no electric vehicle



# ***Analyse some Popular Ev ***

In [ ]:
df5 = pd.read_csv('/content/4-wheeler-EV-carwale.csv')
df5.head()

Sentiments                                            Reviews  Exterior  \
0   Positive  Superb car like as fantastic as petroleum car....         5   
1   Negative  Anti national, worst service, worst customer c...         1   
2   Positive           Super happy with it. The car is too good         4   
3   Positive  Pretty good car, smooth as a glider fast car, ...         5   
4    Nuetral  Price difference between petrol and electronic...         4   

   Comfort  Performance  Fuel_Economy  Value_for_Money      Condition  \
0        4            5             5                5            New   
1        1            1             1                1            New   
2        5            5             5                4            New   
3        5            5             5                5            New   
4        4            5             3                2  Not Purchased   

                    Driven  Rating    Model_name  
0   Few hundred kilometers       5  hyundai kona  
1        Haven't driven it       0  hyundai kona  
2  Few thousand kilometers       5  hyundai kona  
3  Few thousand kilometers       5  hyundai kona  
4        Haven't driven it       3  hyundai kona

### Analyse what attract peoples more to buy Electric vehicle

In [ ]:
df_1 = df5[['Comfort','Performance','Fuel_Economy','Value_for_Money','Rating']]

df_1

Comfort  Performance  Fuel_Economy  Value_for_Money  Rating
0          4            5             5                5       5
1          1            1             1                1       0
2          5            5             5                4       5
3          5            5             5                5       5
4          4            5             3                2       3
..       ...          ...           ...              ...     ...
124        4            4             4                3       4
125        5            5             5                5       5
126        5            5             5                5       5
127        4            4             5                5       5
128        5            5             3                2       5

[129 rows x 5 columns]

In [ ]:
feature = ['Comfort','Performance','Fuel_Economy','Value_for_Money','Rating']
for i in feature:
    df_1[i] = np.where(df_1[i] > 3,1,0)

data = df_1.mean().round(2)
data

<ipython-input-248-7b1fad18d704>:3: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Comfort            0.88
Performance        0.86
Fuel_Economy       0.86
Value_for_Money    0.76
Rating             0.84
dtype: float64

In [ ]:
pd.DataFrame(data, columns = ['percentage']).reset_index()

index  percentage
0          Comfort        0.88
1      Performance        0.86
2     Fuel_Economy        0.86
3  Value_for_Money        0.76
4           Rating        0.84

### To find which car has maximum number of positive reviews

In [ ]:
p1 = df5[df5['Sentiments'] == 'Positive']
p2 = p1.groupby('Model_name')['Sentiments'].count()

In [ ]:
px.bar(p2, title = 'number of positive reviews for each cars')

In [ ]:
r1 = df5[df5['Sentiments'] == 'Negative']
r2 = r1.groupby('Model_name')['Sentiments'].count()
r2 = pd.DataFrame(r2)
r2

Sentiments
Model_name               
hyundai kona            4
tata nexon ev           4
tata tigor ev           4

### To find best car among them

In [ ]:
df5[df5['Comfort'] == 5].groupby('Model_name').count()

Sentiments  Reviews  Exterior  Comfort  Performance  \
Model_name                                                           
hyundai kona           20       20        20       20           20   
tata nexon ev          53       53        53       53           53   
tata tigor ev          13       13        13       13           13   

               Fuel_Economy  Value_for_Money  Condition  Driven  Rating  
Model_name                                                               
hyundai kona             20               20         20      20      20  
tata nexon ev            53               53         53      53      53  
tata tigor ev            13               13         13      13      13

In [ ]:
fd = df5[df5['Comfort'] == 5].groupby('Model_name')['Comfort'].count()
fd

Model_name
hyundai kona     20
tata nexon ev    53
tata tigor ev    13
Name: Comfort, dtype: int64

In [ ]:
px.bar(pd.DataFrame(fd),title = 'Best Car')

# **CONCLUSION :**


*    88% people choose Electric vehicle based on Comfort
*    86% people choose Electric vehicle based on Performance and Fuel Economy
*    76% people choose Electric vehicle because it is value Value for Money
*    84% people given maximum rating for Electric vehicle

TATA nexon ev is the Best EV that people prefers to buy

